base/server/python/pki/server/deployment/__init__.py

```diff
@@ -1314,6 +1351,22 @@ def configure_kra(self, subsystem):
                 serial_dn = 'ou=keyRepository,ou=ranges'
             subsystem.set_config('dbs.serialRangeDN', serial_dn)
 
+        elif key_id_generator == 'legacy2':
+            subsystem.set_config('dbs.beginSerialNumber', '0x1')  # hex
+            subsystem.set_config('dbs.endSerialNumber', '0x10000000')  # hex
+            subsystem.set_config('dbs.serialIncrement', '0x10000000')  # hex
+            subsystem.set_config('dbs.serialLowWaterMark', '0x2000000')  # hex
+            subsystem.set_config('dbs.serialCloneTransferNumber', '0x10000')  # hex
+
+            if key_id_generator == 'legacy2':
+                serial_dn = 'ou=keyRepository,ou=ranges_v2'
+            else:
+                serial_dn = 'ou=keyRepository,ou=ranges'
+            subsystem.set_config('dbs.serialRangeDN', serial_dn)
```

Comment:
We can simplify this code since the generator can only be `legacy2` here. Same thing for the `legacy` case above.